### PR TITLE
[ADR] Handle LDAP product variation via module choice

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: src/adr/0036-against-use-of-conditional-statements-in-guice-modules.md [hunk 1]
```diff
@@ -82,6 +82,9 @@ The following modules perform conditional statements upon injection time:
  - BlobStoreChoosingModule::provideBlobStore: Choice of BlobStore technology: Cassandra, ObjectStorage or Hybrid
  - [Cached blob store](https://github.com/linagora/james-project/pull/3319) represents a similar problem: should the 
  blobStore be wrapped by a caching layer?
+ 
+Cassandra, and Distributed products are furthermore duplicated to offer a version supporting LDAP authentication. JPA
```
Comment (rouazana) [inline src/adr/0036-against-use-of-conditional-statements-in-guice-modules.md:86]:
```suggestion
Cassandra and Distributed products are furthermore duplicated to offer a version supporting LDAP authentication. JPA 
```

File: src/adr/0036-against-use-of-conditional-statements-in-guice-modules.md [hunk 2]
```diff
@@ -82,6 +82,9 @@ The following modules perform conditional statements upon injection time:
  - BlobStoreChoosingModule::provideBlobStore: Choice of BlobStore technology: Cassandra, ObjectStorage or Hybrid
  - [Cached blob store](https://github.com/linagora/james-project/pull/3319) represents a similar problem: should the 
  blobStore be wrapped by a caching layer?
+ 
+Cassandra, and Distributed products are furthermore duplicated to offer a version supporting LDAP authentication. JPA 
+product do not offer LDAP support.
```
Comment (rouazana) [inline src/adr/0036-against-use-of-conditional-statements-in-guice-modules.md:87]:
```suggestion
product does not offer LDAP support.
```

File: src/adr/0036-against-use-of-conditional-statements-in-guice-modules.md [hunk 3]
```diff
@@ -111,4 +114,7 @@ The following conditional statements in guice modules needs to be removed :
  ObjectStorageDependenciesModule::selectBlobStoreBuilder and Cassandra Blob Store Cache conditional statement.
  - [S3 native blobStore implementation](https://github.com/linagora/james-project/pull/3099) along side with S3 endpoints
  support as part of Swift removes the need to select the Object Storage implementation.
- - Follow up work needs to be plan concerning `BlobExportMechanismModule` and `TikaMailboxModule::provideTextExtractor`.
\ No newline at end of file
+ - Follow up work needs to be plan concerning `BlobExportMechanismModule` and `TikaMailboxModule::provideTextExtractor`.
+ 
+We furthermore need to enable a module choice for LDAP on top of other existing products. We should remove LDAP variations
+for LDAP products.
```
Comment (rouazana) [inline src/adr/0036-against-use-of-conditional-statements-in-guice-modules.md:120]: what's the impact on the docker images we are providing?

File: src/adr/0036-against-use-of-conditional-statements-in-guice-modules.md [hunk 4]
```diff
@@ -117,4 +117,5 @@ The following conditional statements in guice modules needs to be removed :
  - Follow up work needs to be plan concerning `BlobExportMechanismModule` and `TikaMailboxModule::provideTextExtractor`.
  
 We furthermore need to enable a module choice for LDAP on top of other existing products. We should remove LDAP variations
-for LDAP products.
\ No newline at end of file
+for LDAP products. Corresponding docker image will be based on their non LDAP version, overriding the `usersrepository.xml`
```
Comment (Arsnael) [inline src/adr/0036-against-use-of-conditional-statements-in-guice-modules.md:120]:
```suggestion
for LDAP products. Corresponding docker images will be based on their non LDAP version, overriding the `usersrepository.xml`
```

**Comment by chibenwa** [comment]
merged

**Pull request closed without merging**